### ISSUE TITLE
Add rule: must define "Promise"

### ISSUE THREAD

**Issue by gajus** [xjamundx/eslint-plugin-promise#2]
A very common error in large project is to use `Promise` without importing it. This can be caught only when testing in an environment that does not declare global variable `Promise` (e.g. IE10). Therefore, I always require that developer import `Promise` to the local scope, e.g.,

```js
import Promise from 'bluebird';
// or
let Promise = require('bluebird');

export default Promise.resolve(null);
```

No need check what is being imported as long as `Promise` variable is defined.

This rule must cause error only when `Promise` variable is used in the document.

**Comment by xjamundx** [comment]
I like this rule!

I don't think it would be too hard to write either (just need to check for a `MemberExpression` where the object name is `Promise` or a `NewExpression` where the callee name is `Promise`)

I'll get to it in about a week unless you want to contribute!

**Comment by gajus** [comment]
> I'll get to it in about a week unless you want to contribute!

My hands are full at the moment.

**Comment by xjamundx** [comment]
Someone created this rule, I'm going to ask if we can bring it into our plugin as well:
https://github.com/colonyamerican/eslint-plugin-cah/blob/master/docs/no-undef-promise.md

**Comment by gajus** [comment]
@xjamundx The issue has been closed. Has the code migration been actioned?

**Comment by xjamundx** [comment]
Finally going to work on this

**Comment by xjamundx** [comment]
Added `no-native` and publshed with `1.3.0`
